src/api/fetchAPI_rest.ts

```diff
@@ -1,27 +1,71 @@
 import { fetchAPI } from "./fetchAPI";
+import { compareFunc, onlyUnique } from "../util/helpers";
 
 export class fetchAPI_rest implements fetchAPI {
     url: string;
-    
+
     constructor(url: string) {
         this.url = url;
     }
 
+    private fetch(url: string) {
+        return fetch(url, { mode: "cors" })
+            .then((res) => res.json())
+            .then((runs) =>
+            runs.map((run) => {
+                // convert date strings to date
+                run.runStartTime = new Date(run.runStartTime);
+                run.attemptStartTime = new Date(run.attemptStartTime);
+                run.attemptStartTimeMillis = new Date(run.attemptStartTime).getTime(); // needed for HistorBarChart
+                run.runEndTime = new Date(run.runEndTime);
+                run.duration =
+                run.runEndTime.getTime() - run.attemptStartTime.getTime();
+                return run;
+            })
+            );
+    }
+
+    groupByWorkflowName(data: any[]) {
+        return data.reduce((group: { [key: string]: any[] }, run) => {
+                if (!group[run.name]) group[run.name] = [];
+                group[run.name].push(run);
+                return group;
+        }, {});
+    }
+
     getWorkflows = () => {
-        return fetch(`${this.url}/workflows`, {mode:'cors'})
-        .then(res => res.json())
-        .catch(err => console.log("Unexpected Server error: ", err))
+        return this.fetch(`${this.url}/workflows`)
+            .then((data) => {
+            const workflows = this.groupByWorkflowName(data);
```

Comment:
This logic should be implemented in the API, if possible with a query to the underlying Database. Otherwise we transfer all runs (easily several thousand) to the frontend, which might be a sever performance problem.